src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -66,6 +66,25 @@
  * r = 5; int t; //violation here
  * </pre>
  * <p>
+ * Note: resource declarations can contain variable definitions
+ * and variable references (from java9).
+ * This check is only applied to variable definitions.
+ * If there are one or more variable references
+ * and one variable definition on the same line in resources declaration,
+ * there is no violation.
+ * The following examples will illustrate difference:
+ * </p>
+ * <pre>
+ * OutputStream stream1 = new PipedOutputStream();
+ * OutputStream stream2 = new PipedOutputStream();
+ * // only one statement with variable definition - no violation
+ * try (stream1; stream2; OutputStream s3 = new PipedOutputStream();)
```

Comment:
make trailing comment "// ok"
and remove "no violation" from above.
It is better to show line where violation will be placed.

please update example below too.

Reply:
done.
Also I changed the resource types to Reader and PipedReader to make line shorter than 100 chars